src/js/pouchdb-common.js

```diff
@@ -0,0 +1,145 @@
+/*
+
+    Provide a component that wraps an individual PouchDB instance.
+
+    https://github.com/GPII/gpii-pouchdb/blob/master/docs/pouchdb.md
+
+ */
+/* eslint-env node */
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var PouchDB = PouchDB || require("pouchdb");
+
+fluid.registerNamespace("gpii.pouch");
+
+gpii.pouch.init = function (that) {
+    that.pouchDb = new PouchDB(that.options.dbOptions);
+    // fluid.log("Pouch instance `" + that.options.dbOptions.name + "` (" + that.id + ") initialized...");
+};
+
+/**
+ *
+ * Call an underlying PouchDb function and fire the named function on completion.
+ *
+ * @param that - The component itself.
+ * @param fnName {String} - The name of the PouchDB function to call.
+ * @param fnArgs {Array} - The arguments (minus the final callback) to pass to the function.
+ * @param eventName {String} - The event to fire on completion.
+ *
+ */
+gpii.pouch.callPouchFunction = function (that, fnName, fnArgs, eventName) {
+    var promise = fluid.promise();
+
+    var wrappedCallback = function (err, results) {
+        if (err) {
+            promise.reject(err);
+            that.events.onError.fire(err);
+        }
+        else {
+            promise.resolve(results);
+            that.events[eventName].fire(results);
+        }
+    };
+
+    // Add the event firer callback to the list of arguments as the final argument.  If there are no args, we create
+    // an empty array first.
+    var fullArgs = fluid.makeArray(fnArgs).concat(wrappedCallback);
+    that.pouchDb[fnName].apply(that.pouchDb, fullArgs);
+
+    return promise;
+};
+
+fluid.defaults("gpii.pouch", {
+    gradeNames: ["fluid.component"],
+    dbOptions: {
+        skip_setup: true
+    },
+    events: {
+        onAllDocsComplete: null,
```

Comment:
Please explain further, it seems much safer to me to have separate events.  Having a single event seems like it would need listeners that include code to filter by sub-event.